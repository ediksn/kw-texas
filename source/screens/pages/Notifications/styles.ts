import { StyleSheet } from 'react-native'
import { moderateScale, scale } from 'react-native-size-matters'

export const styles = StyleSheet.create({
  containerBoxStyle: {
    flex: 1
  },
  container: {
    padding: moderateScale(10),
    flex: 1
  },
  iconStyle: {
    width: scale(60),
    height: scale(60),
    marginBottom: scale(25)
  }
})
