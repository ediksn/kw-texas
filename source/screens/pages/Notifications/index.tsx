/** @format */

import React, { useEffect } from 'react'
import { FlatList, SafeAreaView, View } from 'react-native'
import { useBackButtonMinimize } from '~/hooks'
import { getUsrProfileActions, homeActions } from '~/store/actions'
import { styles } from './styles'
import useNotification from './hooks/useNavigation'

export const Notifications = () => {
  const {
    usrId,
    usrProfile,
    posts,
    limitDefault,
    loading,
    renderEvents,
    keyExtractor,
    onRefresh,
    onEndReached,
    emptyNotifications,
    dispatch
  } = useNotification()

  useBackButtonMinimize()

  useEffect(() => {
    if (usrProfile.length === 0) dispatch(getUsrProfileActions.getUsrProfile(usrId))
    dispatch(homeActions.getPosts(limitDefault))
  }, [])

  return (
    <SafeAreaView style={styles.containerBoxStyle}>
      <View style={styles.container}>
        <FlatList
          renderItem={renderEvents}
          contentContainerStyle={styles.containerBoxStyle}
          ListEmptyComponent={emptyNotifications}
          data={posts}
          keyExtractor={keyExtractor}
          refreshing={loading}
          onRefresh={onRefresh}
          onEndReached={onEndReached}
          onEndReachedThreshold={1}
        />
      </View>
    </SafeAreaView>
  )
}
