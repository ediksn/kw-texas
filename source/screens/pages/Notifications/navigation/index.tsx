import React from 'react'
import { useTranslation } from 'react-i18next'
import { Header } from '~/components'
import { NAVIGATION } from '~/constants'
import { StackNavigator, StackScreen } from '~/screens/components/Navigators'
import { Notifications } from '..'
import { styles } from './styles'

const NoitificationsHeader = () => {
  const { t } = useTranslation()

  const titleText = t('components_Events_Your_Events')

  return {
    header: () => {
      return <Header style={styles.container} title={titleText} titleStyle={styles.titleStyle} />
    }
  }
}

const NotificationsStackScreen = () => (
  <StackNavigator>
    <StackScreen options={NoitificationsHeader} name={NAVIGATION.SCREEN.NOTIFICATIONS} component={Notifications} />
  </StackNavigator>
)

export default NotificationsStackScreen
