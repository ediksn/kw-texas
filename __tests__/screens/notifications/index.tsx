import React from 'react'
import renderer from 'react-test-renderer'
import { Notifications } from '~/screens/pages/Notifications/index'

let mockLoading = false

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux')
  return {
    ...ActualReactRedux,
    useDispatch: jest.fn().mockImplementation(() => {
      return jest.fn()
    }),
    useSelector: jest.fn().mockImplementation(() => {
      return jest.fn()
    })
  }
})

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn()
    }),
    useFocusEffect: jest.fn()
  }
})

jest.mock('~/screens/pages/Notifications/hooks/useNavigation', () => {
  return jest.fn(() => ({
    loading: mockLoading
  }))
})

beforeAll(() => {
  mockLoading = false
})

test('renders correctly', () => {
  const emtpyList = renderer.create(<Notifications />).toJSON()
  expect(emtpyList).toMatchSnapshot()
})
