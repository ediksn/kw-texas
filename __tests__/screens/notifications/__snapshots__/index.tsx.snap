// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 15.714285714285714,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flex": 1,
        }
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
